crypto/edhoc-crypto-hacspec/src/lib.rs

```diff
@@ -93,3 +95,29 @@ pub fn p256_ecdh(
     let secret = BytesP256ElemLen::from_seq(&x.to_byte_seq_be());
     secret
 }
+
+pub fn p256_generate_private_key() -> BytesP256ElemLen {
+    let mut private_key = BytesP256ElemLen::new();
+
+    loop {
+        for i in 0..private_key.len() {
+            private_key[i] = U8(rand::thread_rng().gen::<u8>());
+        }
+        if p256_validate_private_key(&ByteSeq::from_slice(&private_key, 0, private_key.len())) {
+            break;
+        }
+    }
+
+    private_key
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_p256_generate_private_key() {
+        let random_key = p256_generate_private_key();
```

Comment:
Since `p256_generate_private_key` returns `BytesP256ElemLen`, which is of length `P256_ELEM_LEN`, this test will always pass, right?

Reply:
You are right. I can't think of other tests to add here. Maybe just remove it?

Reply:
Same here, agreed